### PR TITLE
better support for basic and digest auth at the same endpoint

### PR DESCRIPTION
**Pull request by czue**
related to (and solves) http://manage.dimagi.com/default.asp?183357

read commit by commit

### CODE REVIEW

**Comment by czue** [comment]
@dannyroberts

**Comment by czue** [comment]
:white_check_mark:

File: corehq/apps/domain/tests/test_guess_phone_type.py [hunk 1]
```diff
@@ -1,10 +1,10 @@
 from django.test import SimpleTestCase
-from corehq.apps.domain.decorators import J2ME, guess_phone_type_from_user_agent, ANDROID
+from corehq.apps.domain.auth import J2ME, ANDROID, determine_authtype_from_request, guess_phone_type_from_user_agent
```
Comment (dimagimon) [inline corehq/apps/domain/tests/test_guess_phone_type.py]: E501 line too long (116 > 115 characters)

**Comment by czue** [comment]
updated to tack on a fix for http://manage.dimagi.com/default.asp?183090 (https://github.com/dimagi/commcare-hq/commit/d2c0b823a8d5e3174d795bfa51321a4caf527a3f)

**Comment by czue** [comment]
:white_check_mark: (again)

**Comment by benrudolph** [comment]
lgtm, will let @dannyroberts take a pass since i don't have too much context

**Comment by dannyroberts** [comment]
Thanks, this is a lot more coherent!

**Pull request merged**